app/controllers/offers_controller.rb

```diff
@@ -0,0 +1,32 @@
+class OffersController < ApplicationController
+
+  def new
+    # the following line of code uses method .find on class Product, searches in all the records of the model
+    # looking for the product_id that is stored in params. Returns an error if not found
+    # we use this to make sure we won't store in the db an offer for a product that doesn't exist
+    Product.find(params["product_id"])
+    # the product_id is passed to Offer.new using query string `?product_id=x`
+    # then it's read from params (params is a hash) and assigned to the product_id cell of the instance of class Offer
+    @offer = Offer.new({product_id: params["product_id"]})
+  end
+  def create
+    # we create a new record in the model Offer, assigning to the respective columns the values stored in offer_params
+    # these are passed from the fields in the offer_new view, which are labelled offer_* in html
+    # so we basically tell rails to take all the values from the fields labelled offer that are in params and assign them to the
```

Comment:
technically "offer" is a key of the params hash, that stores the value of another hash with all offer attributes

Reply:
Not sure I understand; if "offer" is a key, the it should have one value. One key and one value. But there are many values we pull from `offer_params`.